r/R/dplyr-functions.R

```diff
@@ -672,6 +672,18 @@ nse_funcs$strptime <- function(x, format = "%Y-%m-%d %H:%M:%S", tz = NULL, unit
   Expression$create("strptime", x, options = list(format = format, unit = unit))
 }
 
+nse_funcs$strftime <- function(x, format = "", tz = "", usetz = FALSE) {
+  if (usetz) {
+    format <- paste(format, "%Z")
+  }
+  if (tz == "") {
+    tz <- Sys.timezone()
+  }
+  unit <- TimestampType__unit(x$type())
+  ts <- Expression$create("cast", x, options = list(to_type = timestamp(unit, tz)))
```

Comment:
Please could you add a comment above these lines explaining why you're casting this?

Also, calling `TimestampType__unit()` directly feels a little unidiomatic - the functions with that syntax tend to only be called from within R6 classes.  @jonkeane - it'd be good to get your thoughts here.

Reply:
Yeah  I was wondering about that too. I'll add a comment.

Reply:
Ah yes, I think I see what's going on here: when you're accessing `unit`, we already have some helper methods on the type class so you don't need to call `TimestampType__unit()` directly. What you want for line 682 instead if something like `unit <- x$type$unit()`. 

I'm a little surprised/confused by using `type()` in the code that's here instead of `type`, so it's _possible_ that at this point in evaluation you'll actually need `unit <- x$type()$unit()`, but I haven't been able to construct that in my testing:

```
> library(arrow)
> library(tibble)
> 
> times <- tibble(x = c(lubridate::ymd_hms("2018-10-07 19:04:05"), NA))
> tab <- Table$create(times)
> 
> tab$x$type$unit()
[1] 2
> tab$x$type()$unit()
Error: attempt to apply non-function
```

Reply:
Ha, `unit <- x$type()$unit()` does the trick! (`unit <- x$type$unit()` did not) Thanks!